src/bundle-manifest.ts

```diff
@@ -205,10 +204,9 @@ export function generateSharedDepsMergeStrategy(minEntrypoints?: number):
  * bundle for an application shell.
  */
 export function generateShellMergeStrategy(
-    shell: UrlString, minEntrypoints?: number): BundleStrategy {
-  if (minEntrypoints === undefined) {
-    minEntrypoints = 2;
-  }
+    shell: UrlString, maybeMinEntrypoints?: number): BundleStrategy {
+  const minEntrypoints =
+      maybeMinEntrypoints === undefined ? 2 : maybeMinEntrypoints;
```

Comment:
It seems that in 2.2.0 tsc lost some ability to do local type narrowing. My guess would be a bug involving `arguments`, as this code relied upon being able to determine that `minEntrypoints` wasn't assigned to.